lib/core/utils/preload-cssom.js

```diff
@@ -1,208 +1,106 @@
 /**
- * Make an axios get request to fetch a given resource and resolve
- * @method getExternalStylesheet
- * @param {Object} options an object with properties to configure the external XHR
- * @property {Object} options.resolve resolve callback on queue
- * @property {Object} options.reject reject callback on queue
- * @property {String} options.url string representing the url of the resource to load
- * @property {Object} options.rootNode document or shadowDOM root document for which to process CSSOM
- * @property {Number} options.timeout timeout to about network call
- * @property {Function} options.getStyleSheet a utility function to generate a style sheet for a given text content
- * @property {String} options.shadowId an id if undefined denotes that given root is a shadowRoot
- * @property {Number} options.priority css applied priority
- * @returns resolve with stylesheet object
- * @private
- */
-function getExternalStylesheet(options) {
-	const {
-		resolve,
-		reject,
-		url,
-		rootNode,
-		timeout,
-		getStyleSheet,
-		shadowId,
-		priority
-	} = options;
-	axe.imports
-		.axios({
-			method: 'get',
-			url,
-			timeout
-		})
-		.then(({ data }) => {
-			const sheet = getStyleSheet({
-				data,
-				isExternal: true,
-				shadowId,
-				root: rootNode,
-				priority
-			});
-			resolve(sheet);
-		})
-		.catch(reject);
-}
-
-/**
- * Get stylesheet(s) from shadowDOM
- * @param {Object} documentFragment document fragment node
- * @param {Function} getStyleSheet helper function to get stylesheet object
- * @returns an array of stylesheet objects
- */
-function getSheetsFromShadowDom(documentFragment, getStyleSheet) {
-	return Array.from(documentFragment.children).reduce((out, node) => {
-		const nodeName = node.nodeName.toUpperCase();
-		if (nodeName !== 'STYLE' && nodeName !== 'LINK') {
-			return out;
-		}
-		if (nodeName === 'STYLE') {
-			const dynamicSheet = getStyleSheet({ data: node.textContent });
-			out.push(dynamicSheet.sheet);
-		}
-		if (nodeName === 'LINK' && !node.media.includes('print')) {
-			const dynamicSheet = getStyleSheet({ data: node, isLink: true });
-			out.push(dynamicSheet.sheet);
-		}
-		return out;
-	}, []);
-}
-
-/**
- * Filter a given array of stylesheet objects
- * @param {Array<Object>} styleSheets array of stylesheets
- * @returns an filtered array of stylesheets
+ * Given a rootNode - construct CSSOM
+ * -> get all source nodes (document & document fragments) within given root node
+ * -> recursively call `loadCssom` to resolve styles
+ *
+ * @method preloadCssom
+ * @memberof `axe.utils`
+ * @instance
```

Comment:
What does `@instance` mean?

Reply:
http://usejsdoc.org/tags-instance.html

We use `@instance` across the codebase, to mark parent > child namespace relationship.

Reply:
Right, this isn't an instance method tho. This is a static method.

Reply:
You are right mate. I verified how to docs get generated. Removed this.